packages/test/test-end-to-end-tests/src/test/blobs.spec.ts

```diff
@@ -432,3 +434,240 @@ describeNoCompat("blobs", (getTestObjectProvider) => {
         await uploadP;
     });
 });
+
+/**
+ * Validates that unreferenced blobs are marked as unreferenced and deleted correctly.
+ */
+describeNoCompat("Garbage collection of blobs", (getTestObjectProvider) => {
+    // If deleteUnreferencedContent is true, GC is run in test mode where content that is not referenced is
+    // deleted after each GC run.
+    const tests = (deleteUnreferencedContent: boolean = false) => {
+        const gcContainerConfig: ITestContainerConfig = {
+            runtimeOptions: {
+                gcOptions: {
+                    gcAllowed: true, runGCInTestMode: deleteUnreferencedContent, writeDataAtRoot: true,
+                },
+            },
+        };
+
+        let provider: ITestObjectProvider;
+        let container: IContainer;
+        let containerRuntime: ContainerRuntime;
+        let defaultDataStore: ITestDataObject;
+
+        /**
+         * Returns the referenced / unreferenced state of each node if the GC state in summary.
+         */
+        async function getUnreferencedNodeStates() {
+            await provider.ensureSynchronized();
+            const { summary } = await containerRuntime.summarize({
+                runGC: true,
+                fullTree: true,
+                trackState: false,
+            });
+
+            const gcState = getGCStateFromSummary(summary);
+            assert(gcState !== undefined, "GC tree is not available in the summary");
+
+            const nodeTimestamps: Map<string, "referenced" | "unreferenced"> = new Map();
+            for (const [nodePath, nodeData] of Object.entries(gcState.gcNodes)) {
+                // Unreferenced nodes have unreferenced timestamp associated with them.
+                nodeTimestamps.set(nodePath, nodeData.unreferencedTimestampMs ? "unreferenced" : "referenced");
+            }
+            return nodeTimestamps;
+        }
+
+        beforeEach(async function() {
+            provider = getTestObjectProvider();
+            if (provider.driver.type !== "odsp") {
+                this.skip();
+            }
+            const detachedBlobStorage = new MockDetachedBlobStorage();
+            const loader = provider.makeTestLoader({ ...gcContainerConfig, loaderProps: {detachedBlobStorage}});
+            container = await loader.createDetachedContainer(provider.defaultCodeDetails);
```

Comment:
This makes me wonder what would happen if GC was run in a detached state

Reply:
You can run GC in detached state and it should work fine. However, it would not make much sense because the document is in a draft state and all the references may not have been added yet. We could do it when attaching the container because the document should be in a stable state.
There is a work item to investigate this - https://github.com/microsoft/FluidFramework/issues/6505.